apps/member-profile/app/shared/components/resource-form.tsx

```diff
@@ -136,20 +136,75 @@ export function ResourceTagsField({
   const createFetcher = useFetcher<unknown>();
   const listFetcher = useFetcher<SearchTagsResult>();
 
-  const [newTagId, setNewTagId] = useState<string>(id());
   const [search, setSearch] = useState<string>('');
+  const [selectedTags, setSelectedTags] = useState<Record<string, unknown>>({});
 
   useEffect(() => {
     listFetcher.load('/api/tags/search');
   }, []);
 
   const tags = listFetcher.data?.tags || [];
 
-  function reset() {
+  function onTagCreated(newTag: unknown) {
+    createFetcher.submit(
+      { id: newTag.id, name: newTag.label },
```

Comment:
I think there may be some type-related issues here because since the newTag type is unknown, the compiler doesn't know if newTag has the attributes of "id" and "label". It seems this wasn't an issue before because in the simpler uses of `tag`, the structure of `tag` was still more explicit and hence could be inferred by the compiler. 

I'm curious if it'd be worth defining a type called "Tag" so that `newTag` and `tag` can have more explicit types. This could be done as follows: 
```
interface Tag {
  id: string;
  name: string;
}
```